### PR TITLE
Add forms

### PR DESCRIPTION
**Pull request by Apatakiti**
## Project requirements

- Create a method called current_user in your ApplicationController to make current user data available to all controllers.

    -   It will return the first user from the database.

- Don't worry about proper authentication at this time. You will add it in the upcoming projects.

    - Create forms to perform the following functions:
    - Creates a Post on behalf of the current_user (use the method that you created in your ApplicationController).
    - Create a comment on behalf of the current_user (use the method that you created in your ApplicationController).

- Allow Users to add likes to Posts.

## NOTE
There is a bug in the action_pack dependencies which might make `rspec spec` throw error.
you can check this directory for fixing the bug and then re `rspec spec`

`C:\Users\Owoeye Olaoluwa\Documents\Blog_rails\vendor\bundle\ruby\3.2.0\gems\actionpack-7.1.2\lib\action_dispatch\http\response.rb    `

**on line 195**

**from**

```   def has_header?(key) = @headers.key?(key)

    def get_header(key) = @headers.[](key)

    def set_header(key, v) = @headers(key, v)

    def delete_header(key) = @headers.delete(key)  
```

**to**
  ``` 
 def has_header?(key)
      @headers.key?(key)
    end

    def get_header(key)
      @headers[key]
    end

    def set_header(key, v)
      @headers[key] = v
    end

    def delete_header(key)
      @headers.delete(key)
    end
```

### CODE REVIEW

**Comment by rloterh** [comment]
Hi @Ademola-coding 

Can you make your ReadMe look more professional
- Let's remove the logo from ReadMe
- Add Future Features section
- Remove placeholder or instruction text at the License section

Let's do this asap. Cheers

File: app/views/posts/show.html.erb [hunk 1]
```diff
@@ -1 +1,29 @@
...
+        </p>
+      </div>
+    <% end %>
+  </div>
+
+  <div>
+  <%= render "likes/add_like" %>
+  </div>
+  <%= render "comments/add_comment" %>
+</section>
```
Comment (rloterh) [inline app/views/posts/show.html.erb:29]: - [x] The form for adding a new post is displaying correctly, however, an attempt to add a sample post is throwing an error. Let's endeavor to improve upon the same:

**Error thrown whiles saving a post:**
image
a new

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Rite, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 
🏁 🌴 🚦 🚗 🏫 🏡 _🏃

## Highlights 👍  
- Good jobs o far on form creation!
- Good ReadMe

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by abdulhamiid — APPROVED** [review]
# 🟢 Approved! 🥳 
Hi @Ademola-coding 👋, 
image
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 🌟

- ✔️ Descriptions are descriptive
- ✔️ Linters checks are passing
- ✔️ Use preprocessed HTML files with embedded Ruby code.

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**